### ISSUE TITLE
Determine whether order of AuthInfo fields has semantic meaning, and what that meaning is

### ISSUE THREAD

**Issue by perotinus** [kubernetes/cluster-registry#26]
Migrated from a question previously in [the cluster API doc](https://docs.google.com/document/d/1Oi9EO3Jwtp69obakl-9YpLkP764GZzsz95XJlX1a960/edit#):

Are the auth configuration objects conceptually ordered? Is the semantic meaning of the ordering defined by the cluster registry itself, or are users free to define their own semantic meaning to the ordering of auth configurations?

**Comment by perotinus** [comment]
I think this will have to be deferred to community practice, and we will have to see what sorts of paradigms the community comes up with for this (if any). The only potential ordering that makes sense to me is preference (i.e., attempt to authenticate using data from the first `AuthProvider`; if it fails, attempt with the second; etc.) but I'm not willing to make that a standard *de jure* with so little usage and no evidence that it actually has practical value.

**Comment by perotinus** [comment]
No longer relevant after #234.
